sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1482,4 +1482,51 @@ export class ApplicationService extends RecordDataModelService<Application> {
       })
       .getOne();
   }
+
+  async validatePIRAndDateOverlap(
```

Comment:
It would be good to some comments explaining why the PIR is being validated and why it matters.
I believe that the idea that we talked in the past was to try to comment as much as possible and goes beyond the technical and explain a bit more the business part.
For instance, I do know what the query is doing just looking it, but maybe someone not used to the code will not know why we are looking for an offering id that is null, that it means that there is PIR and that means that we do not have sure about the rea start date and end date of the application.

Reply:
Comment added.